src/Engine/GameAction.java

```diff
@@ -80,7 +80,7 @@ public AttackAction(GameMap gameMap, Unit actor, Path movePath, XYCoord atkLoc)
       if( isValid )
       {
         moveLocation = new XYCoord(movePath.getEnd().x, movePath.getEnd().y);
-        defender = gameMap.getLocation(attackLocation).getResident();
+        defender = gameMap.getLocation(attackLocation).getResident(gameMap);
```

Comment:
ugh... this line hurts to read a little bit. I have to pass GameMap into GameMap to see what's in GameMap?